src/app/modules/myinfo/myinfo.types.ts

```diff
@@ -0,0 +1,66 @@
+import { LeanDocument } from 'mongoose'
+
+import { ISpcpMyInfo } from '../../../config/feature-manager'
+import {
+  AuthType,
+  Environment,
+  IFieldSchema,
+  IFormSchema,
+  IMyInfo,
+  MyInfoAttribute,
+} from '../../../types'
+import { ProcessedFieldResponse } from '../submission/submission.types'
+
+export interface IMyInfoServiceConfig {
+  spcpMyInfoConfig: ISpcpMyInfo
+  nodeEnv: Environment
+  appUrl: string
+}
+
+export interface IMyInfoRedirectURLArgs {
+  formId: string
+  formEsrvcId: string
+  requestedAttributes: MyInfoAttribute[]
+}
+
+export interface IPossiblyPrefilledField extends LeanDocument<IFieldSchema> {
+  fieldValue?: string
+}
```

Comment:
this looks helpful - do you think other modules could have use for it?

Reply:
I don't think so actually, fields in the backend shouldn't have a `fieldValue` other than for MyInfo. it might come in useful for the frontend one day but we can always move it into a shared folder at that point.